src/internal_events/nginx_metrics.rs

```diff
@@ -65,10 +94,24 @@ pub struct NginxMetricsStubStatusParseError<'a> {
 
 impl<'a> InternalEvent for NginxMetricsStubStatusParseError<'a> {
     fn emit_logs(&self) {
-        error!(message = "NginxStubStatus parse error.", endpoint = %self.endpoint, error = ?self.error)
+        error!(
+            message = "NginxStubStatus parse error.",
+            endpoint = %self.endpoint,
+            error = %self.error,
+            error_type = "parse_failed",
+            stage = "receiving",
+        );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "endpoint" => self.endpoint.to_owned(),
+            "error" => self.error.to_string(),
+            "error_type" => "parse_failed",
+            "stage" => "processing",
```

Comment:
Should the `stage` here be `receiving` to match the associated log? (or vice versa)

Reply:
you impress me to catch those 😅 thank you!